### ISSUE TITLE
Paper #190: Code Recommendation for Exception Handling

### ISSUE THREAD

**Issue by kstolee** [researchart/fse20#78]
This issue is for discussion related to PR #48 

Authors are applying for the *available* badge.

Author contact: @tamnguyencs

**Comment by kstolee** [comment]
# Authors

Tam The Nguyen, Phong Minh Vu, Tung Thanh Nguyen
Auburn University, Auburn, Alabama, United States

**Comment by art-reviewer** [comment]
[Paper](https://github.com/tamnguyencs/fse20/blob/77f789cedf8e35b8a37cbfe796460ac7005529ec/submissions/available/fse20main-p190-p/esecfse2020-paper190.pdf)

**Comment by AEreviewer** [comment]
cc @tamnguyencs

I have used the provided guidelines to check the different aspects of the artifact. Overall, the artifact contain the input dataset used for the evaluation presented in the paper. The artifact satisfies to the requirements for the Available badge. Some additional documentation on what is contained in the different sub-folders and in the `.csv` files would be nice to allow other researchers to reuse the dataset. 

- [x] Available: The artifact is placed on a publicly accessible archival repository and has a DOI.

Comments:

- The artifact is available online with a DOI.

- The artifact is not documented. Some additional documentation on what is contained in the different sub-folders and in the `.csv` files would be nice to allow other researchers to reuse the dataset. 

- The artifact seems to contain only the input dataset used for the evaluation. There are no results (which I guess is fine since the authors do not apply for a Functional badge).

**Comment by FSEArtifactReviewer835** [comment]
Requested Badge: Available

The artifact consists of a plugin and a dataset set, provided via archival service.  I was able to install the plugin to IntelliJ and Android Studio successfully.  The key commands on MACOSX are slightly different from the reported ones (Command-Alt instead of Control-Alt), but I was able to get the plugin to make recommendations.  It would also be nice if there was a way to activate the plugin from a menu (e.g. when right-clicking on code) instead of just a hotkey.

As for the dataset, it includes studied bugs (before and after), the project lists used, and metadata about the bugs allowing them to be linked back to their associated commits.  I checked a few of them and they aligned (between GitHub, the CSV file, and the before and after code).  However, I am somewhat confused about the numbers of reported bugs. In the paper, 1000 bugs are mentioned (437 if we go by handling fixes, or 734 if the tool is set to the strictest mode), but in the CSV file there are 750 rows, and in the before and after sections there are 611 bug directories that I could count.  

I think this artifact can qualify for the Available badge, but I would like the authors to address the differences in the reported number of bugs studied in the paper, the CSV file, and the code changes directory.  Was there a reason that some of the projects had to be left out of the dataset?

Basic Requirements:
CONTACT.md - Present, author name, emails, and GitHub id present
README.md - Present
INSTALL.md - Present
index.md - Present, contains the DOI
STATUS.md - Present, describes the requested badge and justification for why they should receive it.
LICENSE.md - Present, Creative Commons 4.0
Accepted Paper - Present

Comments on dataset details:
exceptionbugs.csv - Associated information with the bugs fixes. 750 examples.

FDroidProjects.csv - Project information used for all the projects in the second part of the evolution.

projects.csv - Projects used to extract exception bug fixes.

code changes - A directory of bug fixes on exception handling code with both before and after versions.  There are 611 bug samples.  I've checked a few examples here, and the number ids align with the row id in the exceptionbugs.csv, and the diffs align with the reported examples on Github.  However, I'm a little confused on the numbers reported in the dataset.  Where are the missing examples from the 750 in the CSV file?  From what I can tell, some projects like "Wordpress-Android" were left out (rows 214-291).  Was there a reason for this?

**Comment by kstolee** [comment]
Thanks for the reviews @FSEArtifactReviewer835 and @AEreviewer . I'll recommend the ```available``` badge and request that @tamnguyencs address the reviewer comments in the artifact. 

@tamnguyencs can you make the requested changes? Especially the following from @FSEArtifactReviewer835 : 

> I would like the authors to address the differences in the reported number of bugs studied in the paper, the CSV file, and the code changes directory. Was there a reason that some of the projects had to be left out of the dataset?

and the following from @AEreviewer : 

> Some additional documentation on what is contained in the different sub-folders and in the .csv files would be nice to allow other researchers to reuse the dataset.

**Comment by tamnguyencs** [comment]
Thanks to the reviewers @FSEArtifactReviewer835, @AEreviewer, and @kstolee for the reviews of the artifact. I haven't received the notification for the artifact (the FSE website indicates the notification date is Wednesday, July 01). Should I assume that the paper receives the **available** badge as the recommendation? I will make the requested changes regarding the questions from the reviewer. Thank you so much.

**Comment by kstolee** [comment]
@tamnguyencs yes, the "evaldone" label was supposed to be your notification, but I understand that could have been clearer. You received the ```available``` badge.
